tests/core/server/test_rate_limits.py

```diff
@@ -20,14 +21,19 @@ def event_loop():
 class TestRateLimits:
     @pytest.mark.asyncio
     async def test_too_many_messages(self):
+        # The new_*_count values presume that these tests will only span one minute (rate limit)
+        # boundary, not two.
+        new_transaction_count = math.ceil(rate_limits_tx[ProtocolMessageTypes.new_transaction] * 1.1)
```

Comment:
It should be less than 1 right? otherwise it will fail on line 33

Reply:
Yeah, I read over the assert inside the loop in my quick start here.  I was thinking the first loop was just a setup but it is asserting lack of rate limiting.  I'll make another plan.

I think the proper solution is to have the test have control over the time.  We can give a callable to the limiter and then move time forward as we wish in the test.  The default callable can be `time.time` so application code gets the present functionality.  Also, this now appears to really be four separate tests.

Thanks.